src/voting-tools-rs/src/testing/test_api/fake.rs

```diff
@@ -0,0 +1,154 @@
+use crate::data_provider::DataProvider;
+use crate::model::{
+    Delegations, Reg, RegoMetadata, RegoSignature, RewardsAddr, Signature, SlotNo, StakeVKey, TxId,
+    VotingPurpose,
+};
+use bigdecimal::BigDecimal;
+use cardano_serialization_lib::address::Address;
+use cardano_serialization_lib::crypto::{Ed25519Signature, PublicKey};
+use cardano_serialization_lib::metadata::{TransactionMetadatum, TransactionMetadatumLabel};
+use cardano_serialization_lib::utils::Int;
+use mainnet_lib::DbSyncInstance;
+use std::collections::HashMap;
+use std::str::FromStr;
+
+/// Mock db provider based on [`DbSyncInstance`] struct from [`mainnet_lib`] project.
+/// In essence struct keep data in memory and provides query for voting tools logic
+#[derive(Debug)]
+pub struct MockDbProvider {
+    db_sync_instance: DbSyncInstance,
+}
+
+impl DataProvider for MockDbProvider {
+    fn vote_registrations(
+        &self,
+        lower: Option<SlotNo>,
+        upper: Option<SlotNo>,
+    ) -> color_eyre::Result<Vec<Reg>> {
+        let lower = if let Some(lower) = lower {
+            #[allow(clippy::cast_sign_loss)]
+            Some(lower.into_i64()? as u64)
+        } else {
+            None
+        };
+
+        let upper = if let Some(upper) = upper {
+            #[allow(clippy::cast_sign_loss)]
+            Some(upper.into_i64()? as u64)
+        } else {
+            None
+        };
+
+        let filtered_regs = self
+            .db_sync_instance
+            .query_voting_transactions_with_bounds(lower, upper);
+
+        let mut tx_id = 0;
+
+        Ok(filtered_regs
+            .iter()
+            .map(|(_block0, registrations)| {
+                registrations
+                    .iter()
+                    .map(|r| {
+                        let metadata = r.get(&TransactionMetadatumLabel::from(61284u32)).unwrap();
```

Comment:
Maybe we could pull the `TransactionMetadataumLabel`s into consts (or statics if it's not possible to make them const)?

Reply:
done